src/Jira/Api.php

```diff
@@ -268,6 +288,43 @@ public function transition($issueKey, $params)
     }
 
     /**
+     * transation by step name
+     *
+     *
+     * @param $issueKey like YOURPROJ-22
+     * @param $stepName Step name like 'Done' or 'To Do'
+     * @param $params (array of parameters from JIRA API)
+     * @return mixed
+     */
+    public function transitionByStepName($issueKey, $stepName, $params = array())
```

Comment:
> StepName isn't a jira term, but it's easy to use step name (the name of the button above issue) to navigate. It's just easier to reffer it as text than as ID - even if text could be changed and ID cannot. but it would be more understandable if you would do:
> $jira->updatejira(...);
> $jira->transitionByStepName('Approved');
> $jira->updatejira(..);
> $jira->transitionByStepName('Completed'); 

I think, that `transitionByName($transitionName, $issueKey, $params = array())` would be a better name, because it executes transition by name. The issue on which transition is executed is important, but not so much.

Anyway we're not making a release tomorrow, so we can rename this one later on as well.

P.S.
Please answer to inline comments also inline to keep discussions on separate topics separated.

![2016-02-29_1919](https://cloud.githubusercontent.com/assets/1277526/13402547/719ed684-df19-11e5-9168-8e32f32bf531.png)